Implement log4j2

RamOwen99 opened the pull request:
# Implement Log4j2

- All the messages will display in the console in JSON format

![MicrosoftTeams-image](https://user-images.githubusercontent.com/78741607/114083650-df9a6d00-9874-11eb-8088-c7b59d8fede6.png)

RamOwen99 committed:
Add log4j2 dependency

RamOwen99 committed:
Add JSON template

This is a template to show all the console messages in JSON format

RamOwen99 committed:
Configure log4j2

luisgalindoluna reviewed (approved):
Looks good

ivanflores15 commented on src/main/resources/log4j2.properties:
```diff
@@ -0,0 +1,20 @@
+filters=threshold
+
+filter.threshold.type=ThresholdFilter
+filter.threshold.level=debug
```
is there a reason to left "debug" level?

RamOwen99 replied:
Yes, If I delete the line `filter.threshold.level=debug` the server won't start

ivanflores15 replied:
but there are at leats 5 leves,  what is the reason to use "debug" level?

RamOwen99 replied:
Because with this level we can log what is happening in our development time, but I search that for us is better to use the level `info` because this level is used to log a process that we develop.
If we keep the `debug` level we are going to watch every time a log (e.g. like a server status, our logs, etc).
Would you like that do I change the level?

ivanflores15 replied:
I would recommend create an env variable in order to have control on log level and adding `info` level as default.
As you said, for development purposes it's a good idea have `debug` level but for higher environments a huge amount of logs is not recommended because of performance.
Please apply changes.

RamOwen99 replied:
Ok, let me do those changes

RamOwen99 committed:
Change threshold level to info

Before this level we had the debug level and that level it is more
useful with the higher environments. Adn the logs that we need are to
say if a process well good or bad

EmiliaGuillen-dev reviewed (approved):
looks good
